app/src/main/java/com/jojjnator/kalkylator/viewmodel/CalculatorViewModel.kt

```diff
@@ -7,41 +7,45 @@ import com.jojjnator.kalkylator.CalculatorAction
 
 class CalculatorViewModel : ViewModel() {
 
-    var secondInput = mutableStateOf(0)
+    var secondInput = mutableStateOf("")
         private set
-    var firstInput = mutableStateOf(0)
+    var firstInput = mutableStateOf("")
         private set
     var operator = mutableStateOf("")
         private set
 
-    var calculatedSum = mutableStateOf(0)
+    var calculatedSum = mutableStateOf(0.0)
         private set
 
     fun checkInput(action: CalculatorAction) {
         when (action) {
             is CalculatorAction.Clear -> {
                 operator.value = ""
-                secondInput.value = 0
-                firstInput.value = 0
-                calculatedSum.value = 0
+                secondInput.value = ""
+                firstInput.value = ""
+                calculatedSum.value = 0.0
             }
             is CalculatorAction.DoCalculation -> {
```

Comment:
.DoCalculation is very repetitive, you should be able to extract this to a function or make it shorter by making an enum class